syz-manager/manager.go

```diff
@@ -71,19 +71,103 @@ var (
 	flagTests = flag.String("tests", "", "prefix to match test file names (for -mode run-tests)")
 )
 
+// TODO: replace with interface these dependencies.
+type service struct {
+	rpcserv      *rpcserver.Server
+	reporter     *report.Reporter
+	assetStorage *asset.Storage
+	reproMgr     *reproManager
+	vmPool       *vm.Pool
+
+	dash *dashapi.Dashboard
+	// This is specifically separated from dash, so that we can keep dash = nil when
+	// cfg.DashboardOnlyRepro is set, so that we don't accidentially use dash for anything.
+	dashRepro *dashapi.Dashboard
+}
+
+func newManagerService(
+	cfg *mgrconfig.Config,
+	rpcserverMgr rpcserver.Manager,
+	reproMgrView reproManagerView,
+) (*service, error) {
+	var err error
+
+	var vmPool *vm.Pool
+	if !cfg.VMLess {
+		vmPool, err = vm.Create(cfg, *flagDebug)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	reporter, err := report.NewReporter(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	rpcserv, err := rpcserver.New(cfg, rpcserverMgr, *flagDebug)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create rpc server: %w", err)
+	}
+	if err := rpcserv.Start(); err != nil {
+		return nil, fmt.Errorf("failed to start rpc server: %w", err)
+	}
+	log.Logf(0, "serving rpc on tcp://%v", rpcserv.GetPort())
+
+	var dashRepro *dashapi.Dashboard
+	var dash *dashapi.Dashboard
+	if cfg.DashboardAddr != "" {
+		opts := []dashapi.DashboardOpts{}
+		if cfg.DashboardUserAgent != "" {
+			opts = append(opts, dashapi.UserAgent(cfg.DashboardUserAgent))
+		}
+		dash, err = dashapi.New(cfg.DashboardClient, cfg.DashboardAddr, cfg.DashboardKey, opts...)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create dashapi connection: %w", err)
+		}
+		dashRepro = dash
+		if cfg.DashboardOnlyRepro {
+			dash = nil
+		}
+	}
+
+	var assetStorage *asset.Storage
+	if !cfg.AssetStorage.IsEmpty() {
+		assetStorage, err = asset.StorageFromConfig(cfg.AssetStorage, dash)
+		if err != nil {
+			return nil, fmt.Errorf("failed to init asset storage: %w", err)
+		}
+	}
+
+	svc := &service{
+		rpcserv:      rpcserv,
+		reporter:     reporter,
+		assetStorage: assetStorage,
+		dash:         dash,
+		dashRepro:    dashRepro,
+		vmPool:       vmPool,
+	}
+	if vmPool == nil {
+		return svc, nil
+	}
+
+	svc.reproMgr = newReproManager(reproMgrView, vmPool.Count()-cfg.FuzzingVMs, cfg.DashboardOnlyRepro)
+
+	return svc, nil
+}
+
 type Manager struct {
+	impl            *service
```

Comment:
This makes all access longer as `mgr.impl.` and the split between fields that are embed in Manager and moved to impl is unclear.
Is this change necessary?